pkg/reconciler/isbsvc/watcher.go

```diff
@@ -0,0 +1,106 @@
+package isbsvc
+
+import (
+	"context"
+	"fmt"
+
+	"go.uber.org/zap"
+	appv1 "k8s.io/api/apps/v1"
+	"k8s.io/client-go/util/workqueue"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+
+	dfv1 "github.com/numaproj/numaflow/pkg/apis/numaflow/v1alpha1"
+)
+
+// StatefulSetStatusHandler updates the InterStepBufferService status based on the StatefulSet status
+func StatefulSetStatusHandler(mgr manager.Manager, logger *zap.SugaredLogger) handler.EventHandler {
```

Comment:
Instead of doing the operations in the `EventHandler`, we should handle them in `Reconcile()`. 

The purpose of interface `EventHandler` is to `enqueue`.

```
// EventHandler enqueues reconcile.Requests in response to events (e.g. Pod Create).  EventHandlers map an Event
// for one object to trigger Reconciles for either the same object or different objects - e.g. if there is an
// Event for object with type Foo (using source.KindSource) then reconcile one or more object(s) with type Bar.
```

Reply:
Made the changes accordingly